sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -181,3 +183,31 @@ def from_dict(cls, event):
                     " The `source` and `type` params are required."
                     )
         return event_obj
+
+    @staticmethod
+    def _get_bytes(obj):
+        """Event mixin to have methods that are common to different Event types
+        like CloudEvent, EventGridEvent etc.
+        """
+        try:
+            # storage queue
+            return json.loads(obj.content)
+        except ValueError:
+            raise ValueError(
+                "Failed to retrieve content from the object. Make sure the "
+                 + "content follows the CloudEvent schema."
+                )
+        except AttributeError:
+            # eventhubs
+            try:
+                return json.loads(next(obj.body))[0]
+            except KeyError:
```

Comment:
Is it possible an IndexError could be raised here?

Reply:
And in what case would it raise a KeyError?
Do we also need different handling for a StopIteration error? In that scenario, it meant that we had been passed the right object, but the payload was not valid - so I would expect that to then raise a ValueError, no?

Reply:
```Is it possible an IndexError could be raised here?```

We always pass our events as a list in the send method. Even if the user passes a single event, we internally convert it into a list. So, there shouldn't be an index error

```And in what case would it raise a KeyError?```

A keyerror is raised when we try to subscript a SB message with 0. essentially, service bus message is a dict. which is fetched by `json.loads(next(obj.body))`. So doing a `json.loads(next(obj.body))[0]` will give you a key error.

Reply:
```Do we also need different handling for a StopIteration error```
No - uamqp does't throw stop iteration https://github.com/Azure/azure-uamqp-python/blob/master/uamqp/message.py#L1159-L1162

Reply:
Interesting - so if the Event body is an empty list - calling `next` on it doesn't raise? What does it do?

Reply:
```
m = message.Message(body= b'', body_type=uamqp.MessageBodyType.Data)
for data in m.get_data():
  print(data)

>>> b''
```